### PR TITLE
Add technical documents and improve readme

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/mono/Embeddinator-4000/issues/160

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,47 +24,42 @@ and Java (Android and regular Java), across Windows, Linux and macOS platforms.
...
 ## Community
 
 Feel free to join us at our [#managed-interop](https://gitter.im/managed-interop) Gitter discussion channel.
 
 ## Building
 
-Clone this repository and initialize/update submodules as well as solution depends on them.
+- Clone this repository 
+- Initialize/update submodules: `git submodule update --recursive --init`
+- Open the solution file `Embeddinator-4000.sln` with Visual Studio or Xamarin Studio
```
Comment (dalexsoto) [inline README.md]: Should we still mention Xamarin Studio? I think we want Visual Studio for Mac instead

File: README.md [hunk 2]
```diff
@@ -24,47 +24,42 @@ and Java (Android and regular Java), across Windows, Linux and macOS platforms.
...
 
 Feel free to join us at our [#managed-interop](https://gitter.im/managed-interop) Gitter discussion channel.
 
 ## Building
 
-Clone this repository and initialize/update submodules as well as solution depends on them.
+- Clone this repository 
+- Initialize/update submodules: `git submodule update --recursive --init`
+- Open the solution file `Embeddinator-4000.sln` with Visual Studio or Xamarin Studio 
+- Build (F7)
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply README.md]: F7? I thought it was F6, but then Ctrl+B in recent VS. I would just put “hit build”.

File: ProjectStructure.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - nuget generation is here as well, and shared by all platforms.
+- **docs** - External user facing documentation
+- **external** - External libraries used, pulled in as submodules, which include:
+   - IKVM - Used by Objective-C backend
+   - CppSharp - Used by Android/C backend 
+- **objcgen** - Makefiles and code for the Objective-C backend
+- **packages** - External libraries used, pulled in as nugets, which include:
+   - Mono.Cecil - Used by Java/C backend
+   - NUint - Powering unit tests
+   - A large number of Xamarin.Android support libraries used by Java backend
```
Comment (jonathanpeppers) [inline ProjectStructure.md]: These are mostly for tests, making sure c# can use the support libraries for “embeddinated” code. Is that worth mentioning?

File: objcgen/Internals.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+   -  This is the first, and best, stage where items can be removed from binding if they require features not yet supported.
+   -  A number of mappings are done from C# to Objective-C to make APIs more friendly, such as nicely exposing subscripting.
+   -  Items are all OrderBy'ed to force generation to produce consisently ordered output later.
+   -  As each catagory of items is handled (types, methods, etc) a [postprocessor](objcgenerator-postprocessor.cs) walks the data.
+-  This [postprocessor](objcgenerator-postprocessor.cs) analyzes each catagory looking for items that will cause trouble later in generation\compilation or produce suboptimal bindings:
+	- Names that will produce identical selectors or shadow important pre-existing Objective-C selectors
+		- Duplication detection is done via the [Type Mapper](TypeMapper.cs).
+	- [Operator Overloads](OperatorOverloads.cs) can often be exposed in more friendly names that `op_Addition` and are renamed where possible. Where both "friendly" named and operator methods exist, we expose only one copy.
```
Comment (rolfbjarne) [inline objcgen/Internals.md]: typo: `that` => `than`.

File: objcgen/Internals.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	- [SourceWriters](sourcewriter.cs) are created for headers/private headers/implementations to buffer text until it is written to disk and readably handle indentation.
+	- After writing the standard introduction parts to each file, each assembly is procssed in turn.
+	- Each Enum/Protocol/Type/Extension from that assembly is then generated in turn, each from a GenerateFoo method. 
+		- Some but not all Generation methods depend on "helpers" such as [ProtocolHelper](protocolhelper.cs) which help generate correct code.
+		- [NameGenerator](NameGenerator.cs) contains the mapping between C# and Objective-C names for types/arguments.
+- If compilation is requested then the [Embedder's](embedder.cs) Compile () generates and executes clang invocations.
+    - Special post processing occurs in some target types, frameworks for example, and may involve moving files / lipo / etc
+
+
+### Areas for improvements (Update this docuemnt if fixed)
```
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply objcgen/Internals.md]: typo: `docuemnt`

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chamons) [reply README.md]: I'm just removing the hotkey bit. People should know how to build to have a hope of using E4K.

File: objcgen/Internals.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (chamons) [reply objcgen/Internals.md]: Moved to https://github.com/mono/Embeddinator-4000/issues/647

File: objcgen/Internals.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+## Objective-C Generator Internals
+
+Code generators by their nature of generating code that will later be compiled can be difficult to trace through and understand. 
+
+The Objective-C backend covers a number of different platforms (macOS, iOS, tvOS, etc) and packaging techniques (static library, dylib, frameworks).
```
[the remark this answers is not in the document]
Comment (chamons) [reply objcgen/Internals.md]: Yes, I believe so. Let me rewrite to make that clear.

**Pull request merged**